src/main/java/com/kmbl/InventoryManagementService/services/InventoryService.java

```diff
@@ -77,4 +77,52 @@ public void freeInventory(CancelOrderMessage message) throws ServiceException {
     public void deleteInventory(String id) {
         inventoryRepository.deleteById(id);
     }
+
+
+    public ResponseBody updateInventoryforOrder(List<OrderRequestBody> orderRequestItems) {
+        synchronized (this) {
+            InventoryOrderStatus overallStatus = InventoryOrderStatus.FAILED_ORDER;
+            ResponseBody response = new ResponseBody();
+            for (int i = 0; i < orderRequestItems.size(); i++) {
+                Optional<Inventory> tempInventory = inventoryRepository.findBySellerIdAndProductId(orderRequestItems.get(i).getSellerID(),orderRequestItems.get(i).getProductID());
```

Comment:
Add a validation if sellerId is null or ProductId is null, the service will throw exception